src/com/atsebak/embeddedlinuxjvm/commandline/AppCommandLineState.java

```diff
@@ -221,6 +215,24 @@ protected JavaParameters createJavaParameters() throws ExecutionException {
         javaParams.setWorkingDirectory(project.getBasePath());
         javaParams.getProgramParametersList().addParametersString(configuration.getOutputFilePath());
 
+        if (!javaParams.getModuleName().isEmpty()) {
```

Comment:
You don't have a unit test that currently tests this class/method, I would gladly update it if there was one. The condition will be true if there is a module specified in the configuration form, if it is false, it will act as before (load all modules).